base/base/LineReader.cpp

```diff
@@ -109,10 +109,10 @@ void LineReader::Suggest::addWords(Words && new_words)
         std::lock_guard lock(mutex);
         addNewWords(words, new_words, std::less<std::string>{});
         addNewWords(words_no_case, new_words_no_case, NoCaseCompare{});
-    }
 
-    assert(std::is_sorted(words.begin(), words.end()));
-    assert(std::is_sorted(words_no_case.begin(), words_no_case.end(), NoCaseCompare{}));
+        assert(std::is_sorted(words.begin(), words.end()));
+        assert(std::is_sorted(words_no_case.begin(), words_no_case.end(), NoCaseCompare{}));
```

Comment:
@azat I moved your assertions under mutex but take a look - may be there is another reason for failed assertion (I looked at your PR and haven't found any other possible reason): 
```
clickhouse: ../base/base/LineReader.cpp:115: void LineReader::Suggest::addWords(LineReader::Suggest::Words &&): Assertion `std::is_sorted(words_no_case.begin(), words_no_case.end(), NoCaseCompare{})' failed.\r\nAborted (core dumped)\r\nroot@6c563c941ab2:/# "
```

This is from the failed test in this PR on the previous commit (`01065_window_view_event_hop_watch_bounded`)

Reply:
I guess you are right, it is possible that addWords() can be called in parallel:
- one from the initial loading thread
- and one via client interaction

And indeed this can happen for tests/queries/0_stateless/02049_clickhouse_local_merge_tree.expect too since it does not wait until the completion will be loaded (like some other tests for client)

Reply:
> And indeed this can happen for tests/queries/0_stateless/02049_clickhouse_local_merge_tree.expect too since it does not wait until the completion will be loaded (like some other tests for client)

thanks - this suggestion helped me to notice that `disable_suggestion` argument does not work for clickhouse-local.